valhalla/sif/dynamiccost.h

```diff
@@ -247,9 +247,10 @@ class DynamicCost {
     // you have forward access for the mode you care about
     // you dont care about what mode has access so long as its forward
     // you dont care about the direction the mode has access to
-    return (edge->forwardaccess() & access_mask_) ||
-           (ignore_access_ && (edge->forwardaccess() & baldr::kAllAccess)) ||
-           (ignore_oneways_ && (edge->reverseaccess() & access_mask_));
+    return ((edge->forwardaccess() & access_mask_) ||
+            (ignore_access_ && (edge->forwardaccess() & baldr::kAllAccess)) ||
+            (ignore_oneways_ && (edge->reverseaccess() & access_mask_))) &&
+           (edge->use() != baldr::Use::kConstruction);
```

Comment:
return `false` in `IsAccessible` method for roads under construction.

Probably this is not the best place for it. The only reason I added this check here - is to avoid copying. This method is used in every costing model to check if an edge is routable or not.

Reply:
this seems fine to me.

Reply:
hmm. looks like this code may break backward compatibility of valhalla tiles and router. I mean, if you have old version of valhalla router but trying to use new tiles, you might get a route through constructions, because constractions have non-zero access

Reply:
so, probably we should turn off access for constructions

Reply:
updated

Reply:
in my opinion, if the person wants to ignore access they probably want to allow roads under construction

Reply:
I can imagine usecases in real life when roads with forbidden access were actually passed, it's physically may happen. In case of roads under construction it can't be true. Another point is that these changes save current behavior for routing.

Reply:
it totally can be true though, especially when OSM hasnt been yet updated to reflect reality. anyway i was only advocating for when `ignore_access == true` then `custruction` roads could be used, which shouldnt break current behavior. for now though i think its fine the way you have it, just to me its something that is not intuitive when you specify `ignore_access` in your request